src/lib/stitching/exchange/transformers/replaceCommerceDateTimeType.ts

```diff
@@ -0,0 +1,90 @@
+import { Transform } from "graphql-tools"
+import {
+  GraphQLSchema,
+  GraphQLObjectType,
+  GraphQLInterfaceType,
+  GraphQLFieldConfigMap,
+} from "graphql"
+import {
+  visitSchema,
+  VisitSchemaKind,
+  TypeVisitor,
+} from "graphql-tools/dist/transforms/visitSchema"
+import {
+  createResolveType,
+  fieldToFieldConfig,
+} from "graphql-tools/dist/stitching/schemaRecreation"
+import dateField from "schema/fields/date"
+
+type TypeWithSelectableFields =
+  | GraphQLObjectType<any, any>
+  | GraphQLInterfaceType
+
+export class ReplaceCommerceTimeDateType implements Transform {
+  public transformSchema(schema: GraphQLSchema): GraphQLSchema {
+    const newSchema = visitSchema(schema, {
+      [VisitSchemaKind.OBJECT_TYPE]: ((type: GraphQLObjectType<any, any>) => {
+        const fields = this.transformFields(type)
+        return (
+          fields &&
+          new GraphQLObjectType({
+            fields,
+            name: type.name,
+            description: type.description,
+            astNode: type.astNode,
+            extensionASTNodes: type.extensionASTNodes,
+            isTypeOf: type.isTypeOf,
+            interfaces: type.getInterfaces(),
+          })
+        )
+      }) as TypeVisitor,
+
+      [VisitSchemaKind.INTERFACE_TYPE]: ((type: GraphQLInterfaceType) => {
```

Comment:
Might be we can just visit `FIELD_TYPE` and return `{...fieldConfig, ...dateField}` if it is a `CommerceDateTime`? That would dramatically simplify this code if possible 😅

Reply:
ooh how would that be? i did `...dateField` but in `transformFields` take a look but im all for any possibility of simplifying this.

Reply:
looks like it's not possible to visit field definitions directly with the visitSchema helper, so my idea wouldn't work. Oh well. What you've got already is easy enough to follow :+1: